test/ParameterStoreBuildWrapperPluginTest.groovy

```diff
@@ -25,6 +33,154 @@ class ParameterStoreBuildWrapperPluginTest {
+
+            plugin.apply(stage)
+
+            verify(stage, times(2)).decorate(anyString(), eq(expectedClosure))
+        }
+
+        @Test
+        void decorateTheTerraformEnvironmentStageWhenGlobalParametersSet() {
+            String organization = "MyOrg"
+            String repoName     = "MyRepo"
+            String environment  = "MyEnv"
+            String path         = '/someOtherPath/'
+            Map apo             = [path: "/${organization}/${repoName}/${environment}/", credentialsId: "${environment.toUpperCase()}_PARAMETER_STORE_ACCESS"]
+            Map gp              = [path: path]
+            def firstClosure    = { -> }
+            def secondClosure   = { -> }
+            configureJenkins(repoName: repoName, organization: organization)
+
+            TerraformEnvironmentStage stage         = mock(TerraformEnvironmentStage.class)
+            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+
+            doReturn(environment).when(stage).getEnvironment()
+            doReturn(apo).when(plugin).getEnvironmentParameterOptions(environment)
+            doReturn(firstClosure).when(plugin).addParameterStoreBuildWrapper(gp)
+            doReturn(secondClosure).when(plugin).addParameterStoreBuildWrapper(apo)
+
+            plugin.withGlobalParameter(path)
+            plugin.apply(stage)
+
+            verify(stage, times(2)).decorate(anyString(), eq(firstClosure))
+            verify(stage, times(2)).decorate(anyString(), eq(secondClosure))
+        }
+
+    }
+
+    public class GetEnvironmentParameterOptions {
+        @After
+        public void reset() {
+            Jenkinsfile.instance = null
+            ParameterStoreBuildWrapperPlugin.reset()
+        }
+
+        private configureJenkins(Map config = [:]) {
+            Jenkinsfile.instance = mock(Jenkinsfile.class)
+            when(Jenkinsfile.instance.getStandardizedRepoSlug()).thenReturn(config.repoSlug)
+            when(Jenkinsfile.instance.getRepoName()).thenReturn(config.repoName ?: 'repo')
+            when(Jenkinsfile.instance.getOrganization()).thenReturn(config.organization ?: 'org')
+            when(Jenkinsfile.instance.getEnv()).thenReturn(config.env ?: [:])
+        }
+
+        @Test
+        void validPathBasedOnEnvironment() {
+            String organization = "MyOrg"
+            String repoName     = "MyRepo"
+            String environment  = "qa"
+            Map expected        = [path: "/${organization}/${repoName}/${environment}/".toString()]
+
+            configureJenkins(repoName: repoName, organization: organization)
+            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
+            doReturn(expected.path.toString()).when(plugin).pathForEnvironment(environment)
+
+            Map actual = plugin.getEnvironmentParameterOptions(environment)
+
+            assertEquals(expected.path, actual.path)
+        }
```

Comment:
I might consider simplifying this.  

1. `expected` doesn't  have to be a Map.  You're putting an expected path into `expected`, then always pulling path out.  You could make a simpler 1-to-1 implementation with: 
```
        @Test
        void validPathBasedOnEnvironment() {
            String organization = "MyOrg"
            String repoName     = "MyRepo"
            String environment  = "qa"
            Map expectedPath = "/${organization}/${repoName}/${environment}/".toString()

            configureJenkins(repoName: repoName, organization: organization)
            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
            doReturn(expectedPath.toString()).when(plugin).pathForEnvironment(environment)

            Map actual = plugin.getEnvironmentParameterOptions(environment)

            assertEquals(expectedPath, actual.path)
        }
```
2. Once you do that, the path itself is irrelevant  to the method under test (getEnvironmentParameterOptions).  Org/Repo - that's not actually known to or used by getEnvironmentParameterOptions.  So  we  could simplify further.
```
        @Test
        void validPathBasedOnEnvironment() {
            String environment  = "qa"
            Map expectedPath = "somepath"
            ParameterStoreBuildWrapperPlugin plugin = spy(new ParameterStoreBuildWrapperPlugin())
            doReturn(expectedPath.toString()).when(plugin).pathForEnvironment(environment)

            Map actual = plugin.getEnvironmentParameterOptions(environment)

            assertEquals(expectedPath, actual.path)
        }
```

Reply:
done